actors/builtin/miner/deadline_assignment.go

```diff
@@ -185,12 +198,22 @@ func assignDeadlines(
 	for _, sector := range sectors {
 		info := dlHeap.deadlines[0]
 
+		if info.maxPartitionsReached {
+			return changes, xerrors.Errorf("maxPartitions limit %d reached for all deadlines", maxPartitions)
+		}
+
 		changes[info.index] = append(changes[info.index], sector)
 		info.liveSectors++
 		info.totalSectors++
 
+		nPartitions := info.totalSectors / partitionSize
+		if nPartitions == maxPartitions {
+			info.maxPartitionsReached = true
```

Comment:
nit: I wouldn't actually have this bool. I'd just check `if info.totalSectors >= partitionSize*maxPartitions { return }`.

Reply:
I'd like to. keep this around  as it makes the code easier to read in all the places this is used.

Reply:
Then we'd need to set it on initialization in case the deadline was already full. But I _really_ think we should get rid of it so we don't forget to set it somewhere.

What about turning it into a helper function on the `deadlineAssignmentInfo` type?

Reply:
This makes sense. Done.